bin/tests/integration/named_metrics_tests.rs

```diff
@@ -0,0 +1,260 @@
+                &store_sqlite_secondary,
+                Some(0f64),
+            );
+        }
+
+        // check store metrics
+        // forwarder store only has QueryStoreMetrics
+        // sqlite store not initialized within example_forwarder.toml
+        let store_file = [("store", "file")];
+        verify_metric(
+            metrics,
+            "hickory_zone_records_total",
+            &store_file,
+            Some(14f64),
+        );
+
+        verify_metric(
+            metrics,
+            "hickory_zone_record_lookups_total",
+            &STORE_FILE_SUCCESS,
+            Some(0f64),
+        );
+        verify_metric(
+            metrics,
+            "hickory_zone_record_lookups_total",
+            &STORE_FILE_FAILED,
+            Some(0f64),
+        );
+        verify_metric(
+            metrics,
+            "hickory_zone_record_lookups_total",
+            &STORE_FORWARDER_SUCCESS,
+            Some(0f64),
+        );
+        verify_metric(
+            metrics,
+            "hickory_zone_record_lookups_total",
+            &STORE_FORWARDER_FAILED,
+            Some(0f64),
+        );
+
+        // sqlite store is not configured within example_forwarder.toml
+        // therefore StoreMetrics for sqlite are not initialized
+
+        // currently this feature returns is NotImpl, only functional with sqlite && dnssec feature
+        // empty metrics available for file store as they are part of PersistentStoreMetrics
+        // migrate to Option within PersistentStoreMetrics ?
+        #[cfg(feature = "__dnssec")]
+        {
+            let store_file_dynamic_added = [("store", "file"), ("operation", "added")];
+            let store_file_dynamic_deleted = [("store", "file"), ("operation", "deleted")];
+            let store_file_dynamic_updated = [("store", "file"), ("operation", "updated")];
+
+            verify_metric(
+                metrics,
+                "hickory_zone_records_dynamically_modified_total",
+                &store_file_dynamic_added,
+                Some(0f64),
+            );
+            verify_metric(
+                metrics,
+                "hickory_zone_records_dynamically_modified_total",
+                &store_file_dynamic_deleted,
+                Some(0f64),
+            );
+            verify_metric(
+                metrics,
+                "hickory_zone_records_dynamically_modified_total",
+                &store_file_dynamic_updated,
+                Some(0f64),
+            );
+        }
+    })
+}
+
+async fn fetch_parse_check_metrics(socket_ports: &SocketPorts) -> Scrape {
+    let prometheus_port = socket_ports.get_v4(ServerProtocol::PrometheusMetrics);
+    let addr = SocketAddr::from((
+        Ipv4Addr::LOCALHOST,
+        prometheus_port.expect("no prometheus_port"),
+    ));
+
+    // the collect interval for process metrics is set to 1000ms
+    // wait to avoid missing the process metrics
+    tokio::time::sleep(Duration::from_millis(1000)).await;
```

Comment:
Nit: import `sleep()`, please, and `from_secs()`.

Reply:
Updated.